aten/src/ATen/native/TensorShape.cpp

```diff
@@ -177,7 +177,12 @@ Tensor & _cat_out_cpu(Tensor& result, TensorList tensors, int64_t dim) {
   // compute the size of the result
   auto result_size = notSkippedTensor.sizes().vec();
   result_size[dim] = cat_dim_size;
-  result.resize_(result_size, first_tensor_mem_format);
+
+  // skip resizing if size of result is same as expected
+  if ((result.sizes() != result_size) || result.suggest_memory_format() != first_tensor_mem_format){
```

Comment:
Why do you need second condition here? It would lead to restriding of result instead of respecting its strides?

Reply:
The reason for second condition is that if result has the standard layout and if `first_tensor_mem_format` is say `channels_last`, then even if they have same shape, their strides would be different. In that case we would want to resize the result.

Sample:
```python
>>> torch.ones((4, 3, 2, 5)).stride()
(30, 10, 5, 1)
>>> torch.ones((4, 3, 2, 5)).contiguous(memory_format=torch.channels_last).stride()
(30, 1, 15, 3)
```

Reply:
According to https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch, if `out` has the correct size it has to be used directly, even if it's strides are not what operation would naturally produce.